docs/guides/engineering/fresh-config.md

```diff
@@ -0,0 +1,78 @@
+# Ensuring Fresh Feature Flag Configurations
+
+When working with feature flags, there are certain use cases, like kill switches, where it can be important to ensure your application always has the most up-to-date configuration. This guide explains how to maintain fresh configurations across different platforms when using Eppo's SDKs.
+
+## Server-Side Applications
+
+Server-side applications automatically handle configuration freshness through built-in polling mechanisms. You can configure the polling interval during initialization of the SDK.
+
+## Client-Side Applications
+
+### JavaScript Client
+For browser-based applications, you have several options to ensure fresh configurations:
+
+1. **Enable Polling**
+```javascript
+const client = new EppoClient({
+    apiKey: 'your-api-key',
+    pollInterval: 30000 // Poll every 30 seconds
+});
+```
+
+2. **Manual Refresh**
+```javascript
+// Force a configuration refresh when needed
+await eppoClient.load();
+```
+
+### Mobile Applications
+Mobile apps can implement similar strategies:
+
+1. **Configure Polling**
+```swift
+// iOS Example
+let client = EppoClient(
+    apiKey: "your-api-key",
+    pollInterval: 30 // seconds
+)
+```
+
+```kotlin
+// Android Example
+val client = EppoClient.builder()
+    .apiKey("your-api-key")
+    .pollInterval(30) // seconds
+    .build()
+```
+
+## Best Practices and Considerations
+
+When implementing polling or real-time updates, consider how configuration changes might affect active user sessions. Specifically,Configuration changes during an active session might create an inconsistent user experience. Consider these strategies:
+  - Reserve changing flag values in a session for only the flags that absolutely need it, like kill switches
+  - Update configurations only between sessions
+  - Implement graceful transitions when flags change
+
+### Real-Time Updates
+For the cases when real-time updates are necessary, we recommend adding a listener for a push from the server to force a refresh:
+
+1. **WebSocket Integration**
+```javascript
+// Example of listening to a WebSocket for configuration updates
+webSocket.on('config-update', async () => {
+    await client.refreshConfiguration();
```

Comment:
I do like this interface though! We should consider something like this as syntatic sugar for reinitializing with same config and force reinit 😅